config/src/test/scala/com/typesafe/config/impl/ConfigSubstitutionTest.scala

```diff
@@ -924,20 +924,24 @@ class ConfigSubstitutionTest extends TestUtils {
 
     @Test
     def substSelfReferenceIndirect() {
+        // this has two possible outcomes depending on whether
+        // we resolve and memoize a first or b first. currently
+        // java 8's hash table makes it resolve OK, but
```

Comment:
the spec specifically describes this situation and says it's undefined. LinkedHashMap's overhead seems like overkill to handle a pathological situation that as far as I know never happens in real configs anyhow...